### ISSUE TITLE
Recursion error in Package Browser

### ISSUE THREAD

**Issue by brickviking** [dolphinsmalltalk/Dolphin#688]
## TL;DR ##
I struck a wee problem in the Package Browser for master branch (commit 1831) after I create a package. This bug applies once I try searching for that package name in the Package Browser or even in the Class/System browser (if a Package Browser would be spawned) regardless of whether or not I've created a class in that package, and *only* if I haven't selected a category ($ or Core) first. The same bug can be triggered on 7.0.57.2 under Windows. I also tried this on 7.0.57.2 under wine and didn't get the same error, as the $ category comes up selected when I started up the Package manager (a difference from running under Windows). Labels are *bug* *UI*, if those are relevant enough.

## To repeat ##
* Open Package Browser, Create new Package "Ambulate" (for example). Note, I did not embed this under the normal hierarchy of /Core/anything or even /Core/Contributions. Save Package to disk, Close Package Browser.

* Open System Browser, create new class, call it anything (for example, "Articulate"),
  select "Packaged", choose "Ambulate", click OK.

* Re-open Package Browser, type in "Ambulate" into the search field. Hit Enter (twice... once to finish the string, once to search on the entered string) and watchTheGoodTimes roll. Once again, I did *not* select a category (either $ or Core) to trigger the debug below. Once I do select one, my new package comes up in the bottom list, and can be searched for, skipping the error.

>11:45:53 a.m., Monday, 19 November 2018: 'Recursion too deep; the stack overflowed.'
ProcessorScheduler>>stackOverflow:
[] in ProcessorScheduler>>vmi:list:no:with:
BlockClosure>>ifCurtailed:
ProcessorScheduler>>vmi:list:no:with:
[] in ListView(ListControlView)>>handlesFromObjects:whenAbsent:
Array(ArrayedCollection)>>do:
ListView(ListControlView)>>handlesFromObjects:whenAbsent:
ListView>>selections:ifAbsent:
[] in ListPresenter(SelectableItemsPresenter)>>selections:ifAbsent:
ListView>>selections:ifAbsent:
ListPresenter(SelectableItemsPresenter)>>selections:ifAbsent:
PackageSelector(SelectableItemsPresenter)>>selections:
PackageSelector>>resetForItems:
[] in PackageSelector(SelectableItemsPresenter)>>selections:
ListView>>selections:ifAbsent:
[] in ListPresenter(SelectableItemsPresenter)>>selections:ifAbsent:
ListView>>selections:ifAbsent:
ListPresenter(SelectableItemsPresenter)>>selections:ifAbsent:
PackageSelector(SelectableItemsPresenter)>>selections:
PackageSelector>>resetForItems:
[] in PackageSelector(SelectableItemsPresenter)>>selections:
ListView>>selections:ifAbsent:
[] in ListPresenter(SelectableItemsPresenter)>>selections:ifAbsent:
ListView>>selections:ifAbsent:
ListPresenter(SelectableItemsPresenter)>>selections:ifAbsent:
PackageSelector(SelectableItemsPresenter)>>selections:
PackageSelector>>resetForItems:
[] in PackageSelector(SelectableItemsPresenter)>>selections:
ListView>>selections:ifAbsent:
[] in ListPresenter(SelectableItemsPresenter)>>selections:ifAbsent:
ListView>>selections:ifAbsent:
ListPresenter(SelectableItemsPresenter)>>selections:ifAbsent:
PackageSelector(SelectableItemsPresenter)>>selections:
PackageSelector>>resetForItems:
[] in PackageSelector(SelectableItemsPresenter)>>selections:
ListView>>selections:ifAbsent:
[] in ListPresenter(SelectableItemsPresenter)>>selections:ifAbsent:
ListView>>selections:ifAbsent:
ListPresenter(SelectableItemsPresenter)>>selections:ifAbsent:
PackageSelector(SelectableItemsPresenter)>>selections:
PackageSelector>>resetForItems:
[] in PackageSelector(SelectableItemsPresenter)>>selections:
ListView>>selections:ifAbsent:
[] in ListPresenter(SelectableItemsPresenter)>>selections:ifAbsent:
ListView>>selections:ifAbsent:
ListPresenter(SelectableItemsPresenter)>>selections:ifAbsent:
PackageSelector(SelectableItemsPresenter)>>selections:
PackageSelector>>resetForItems:
[] in PackageSelector(SelectableItemsPresenter)>>selections:
ListView>>selections:ifAbsent:

## Conclusion ##
I'd love this to be fixed, or tell me what I'm doing wrong. Let me know what other information you might need. Thanks for the attention.

Regards, BrickViking (a.k.a. DrSmokey)
